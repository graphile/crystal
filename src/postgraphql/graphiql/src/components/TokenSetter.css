.root {
  display: flex;
  align-items: center;
  flex: 1;
}

.monospace {
  font-family: 'Consolas', 'Inconsolata', 'Droid Sans Mono', 'Monaco', monospace;
}

.label {
  letter-spacing: 1px;
  font-weight: 700;
  font-size: 12px;
  padding-right: 4px;
}

.token {
  composes: monospace;
  color: #777;
  cursor: pointer;
  font-size: 12px;
  display: block;
  max-width: 200px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.token:hover {
  text-decoration: underline;
}

.button {
  outline: none;
  background: #fdfdfd;
  border-width: 0.5px;
  border-style: solid;
  border-color: #d3d3d3 #d0d0d0 #bababa;
  border-radius: 4px;
  box-shadow: 0 1px 1px -1px rgba(0, 0, 0, 0.13), inset 0 1px #fff;
  display: inline-block;
  margin-left: 6px;
  color: #444;
  cursor: pointer;
  padding: 4px 8px 5px;
  text-decoration: none;
}

.form {
  margin: 0;
  width: 100%;
  padding-right: 24px;
}

.input {
  composes: monospace;
  height: 32px;
  font-size: 12px;
  box-sizing: border-box;
  width: 100%;
  padding: 0 6px;
}
