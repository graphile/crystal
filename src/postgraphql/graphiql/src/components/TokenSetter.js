import React from 'react'
import cn from 'classnames'
import styles from './TokenSetter.css'

class TokenSetter extends React.Component {
  static propTypes = {
    token: React.PropTypes.string,
    setToken: React.PropTypes.func.isRequired,
  }

  state = {
    showInput: false,
    tokenInputValue: '',
  }

  render() {
    return (
      <div className={styles.root}>
        <span className={styles.label}>JWT: </span>
        {this.state.showInput
          ? <Input
              tokenInputValue={this.state.tokenInputValue}
              onInputChange={this.onInputChange}
              handleSubmit={this.handleSubmit}
              toggleInput={this.toggleInput}
            />
          : <Token
              token={this.props.token}
              clearToken={this.clearToken}
              toggleInput={this.toggleInput}
            />
        }
      </div>
    )
  }

  toggleInput = () => {
    const show = !this.state.showInput
    this.setState({ showInput: show })
  }

  clearToken = () => {
    this.setState({ tokenInputValue: '' })
    this.props.setToken(null)
  }

  onInputChange = event => {
    this.setState({ tokenInputValue: event.target.value })
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.setToken(this.state.tokenInputValue)
    this.setState({ showInput: false })
  }
}

const Token = ({
  token,
  toggleInput,
  clearToken,
}) => (
  <div className={styles.root}>
    <span
      id="token"
      onClick={toggleInput}
      className={styles.token}>
      {token ? token : 'Click to set token.'}
    </span>
    <button
      className={styles.button}
      onClick={token ? clearToken : toggleInput}>
      {token ? 'Clear' : 'Set'}
    </button>
  </div>
)

const Input = ({
  handleSubmit,
  onInputChange,
  toggleInput,
  tokenInputValue,
}) => (
  <form
    className={styles.form}
    onSubmit={handleSubmit}>
    <input
      autoFocus
      placeholder="Press enter to submit"
      className={cn(styles.input, 'animated', 'fadeIn')}
      value={tokenInputValue}
      onChange={onInputChange}
      onBlur={toggleInput}
    />
  </form>
)

export default TokenSetter
