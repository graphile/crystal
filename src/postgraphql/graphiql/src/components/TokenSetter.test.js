import React from 'react'
import TokenSetter from './TokenSetter'
import { mountToJson } from 'enzyme-to-json'
import { mount } from 'enzyme'

const TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJwb3N0Z3JhcGhxbCIsInJvbGUiOiJ0ZXN0X2p3dF9yb2xlIn0.26olvkrTeBD4UXTsKf15501wexCr47nx9AWgHIgi8a4'
const setToken = jest.fn()

test('will render component', () => {
  let component = mount(
    <TokenSetter token={TOKEN} setToken={setToken}/>
  )
  expect(mountToJson(component)).toMatchSnapshot()

  component.find('Token').find('span').simulate('click')
  expect(mountToJson(component)).toMatchSnapshot()
})
