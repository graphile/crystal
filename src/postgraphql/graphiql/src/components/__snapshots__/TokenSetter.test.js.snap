exports[`test will render component 1`] = `
<TokenSetter
  setToken={[Function]}
  token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJwb3N0Z3JhcGhxbCIsInJvbGUiOiJ0ZXN0X2p3dF9yb2xlIn0.26olvkrTeBD4UXTsKf15501wexCr47nx9AWgHIgi8a4">
  <div
    className="root">
    <span
      className="label">
      JWT: 
    </span>
    <Token
      clearToken={[Function]}
      toggleInput={[Function]}
      token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJwb3N0Z3JhcGhxbCIsInJvbGUiOiJ0ZXN0X2p3dF9yb2xlIn0.26olvkrTeBD4UXTsKf15501wexCr47nx9AWgHIgi8a4">
      <div
        className="root">
        <span
          className="token"
          id="token"
          onClick={[Function]}>
          eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJwb3N0Z3JhcGhxbCIsInJvbGUiOiJ0ZXN0X2p3dF9yb2xlIn0.26olvkrTeBD4UXTsKf15501wexCr47nx9AWgHIgi8a4
        </span>
        <button
          className="button"
          onClick={[Function]}>
          Clear
        </button>
      </div>
    </Token>
  </div>
</TokenSetter>
`;

exports[`test will render component 2`] = `
<TokenSetter
  setToken={[Function]}
  token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJwb3N0Z3JhcGhxbCIsInJvbGUiOiJ0ZXN0X2p3dF9yb2xlIn0.26olvkrTeBD4UXTsKf15501wexCr47nx9AWgHIgi8a4">
  <div
    className="root">
    <span
      className="label">
      JWT: 
    </span>
    <Input
      handleSubmit={[Function]}
      onInputChange={[Function]}
      toggleInput={[Function]}
      tokenInputValue="">
      <form
        className="form"
        onSubmit={[Function]}>
        <input
          autoFocus={true}
          className="input animated fadeIn"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Press enter to submit"
          value="" />
      </form>
    </Input>
  </div>
</TokenSetter>
`;
